notebooks/12-xarray-advanced.md

```diff
@@ -0,0 +1,686 @@
+    
+__Note:__ Xarray adds a [`groupby_bins`](http://xarray.pydata.org/en/stable/generated/xarray.Dataset.groupby_bins.html#xarray.Dataset.groupby_bins) convenience function for binned groups (instead of each value).
+
+</div>
+
++++
+
+### Let's practice
+
++++
+
+Run this cell before doing the exercises, so the starting point is again the `era5_renamed` data set:
+
+```{code-cell} ipython3
+era5 = xr.open_dataset("./data/era5-land-monthly-means_example.nc")
+mapping = {
+    "sf": "snowfall_m",
+    "sp": "pressure_pa",
+    "t2m": "temperature_k",
+    "tp": "precipitation_m",
+    "u10": "wind_ms"
+}
+era5_renamed = era5.rename(mapping)
+```
+
+<div class="alert alert-success">
+
+**EXERCISE**:
+
+Select the pressure data for the pixel closest to the center of Ghent (lat: -51.05, lon: 3.71) and assign the outcome to a new variable `ghent_pressure`.
+
+Define a Matplotlib `Figure` and `Axes` (respectively named `fig, ax`) and use it to create a plot that combines the yearly average of the pressure data in Gent with the monthly pressure data as function of time for that same pixel as line plots. Change the name of the y-label to `'Pressure (Pa)'` and the title of the plot to `'Pressure (Pa) in Ghent (at -51.05, 3.71)'` (see notebook [visualization-01-matplotlib.ipynb](./visualization-01-matplotlib.ipynb#An-small-cheat-sheet-reference-for-some-common-elements) for more information)
+
+<details><summary>Hints</summary>
+    
+* For the yearly average, actually both `resample(time="Y")` as `groupby("time.year")` can be used. The main difference is the returned dimension after the grouping: `resample` returns a DateTimeIndex, whereas `groupby` returns a dimension called `year` and no longer contains the DateTimeIndex. To combine the data with other time series data, using the DateTimeIndex is preferred (i.e. `resample`). 
+* `fig, ax = plt.subplots()` is a useful shortcut to prepare a Matplotlib Figure. You can add a label `set_label()` and title `set_title()` to the `axes` object.
+
+</details>    
+    
+</div>
+
+```{code-cell} ipython3
+:tags: [nbtutor-solution]
+
+ghent_pressure = era5_renamed.sel(latitude=51.05, longitude=3.71, method="nearest")["pressure_pa"]
+
+fig, ax = plt.subplots(figsize=(18, 6))
+ghent_pressure.plot.line(ax=ax)
+ghent_pressure.resample(time="Y").mean().plot.line(ax=ax)
+
+ax.set_ylabel('Pressure (Pa)')
+ax.set_title('Pressure (Pa) in Ghent (-51.05, 3.71)')
+```
+
+<div class="alert alert-success">
+
+**EXERCISE**:
+    
+Select the precipitation data for the pixel closest to the center of Ghent (lat: -51.05, lon: 3.71) and assign the outcome to a new variable `ghent_precipitation`.
+    
+For the Ghent pixel, calculate the maximal precipitation _for each month of the year_ (1 -> 12) and convert it to mm precipitation.
```

Comment:
It's certainly interesting to look at the maximum precipitation per month, a possibly typical thing to do would be to look at the average (over all years) monthly total precipitation. 

So something like `ghent_precipitation.resample(time="M").sum().groupby("time.month").mean()`

(could also be as a separate exercise, it gives a different yearly pattern)

Reply:
Ah, it's already monthly data, so the resample is of course not needed .. which makes it less different

Reply:
Yeah, a resample / groupby combination is limited with the monthly frequency of the data. I thought we could do something like that with the  average (over all years) seaonsal total precipitation, but season is not supported in resample :-(.